### PR TITLE
[server] Fix layer order rendering when layers are in groups

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #20810

### CODE REVIEW

File: src/server/services/wms/qgswmsrenderer.cpp [hunk 1]
```diff
@@ -2725,6 +2725,8 @@ namespace QgsWms
       }
       else if ( mLayerGroups.contains( nickname ) )
       {
+        // Reverse order of layers form a group
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsrenderer.cpp]: form -> from

File: tests/src/python/test_qgsserver_accesscontrol.py [hunk 2]
```diff
@@ -217,31 +217,9 @@ def _img_diff(self, image, control_image, max_diff, max_size_diff=QSize()):
...
-            with open(os.path.join(tempfile.gettempdir(), image + "_result_diff.png"), "rb") as diff_file:
-                encoded_diff_file = base64.b64encode(diff_file.read())
-                message += "\nDiff:\necho '%s' | base64 -d > %s/%s_result_diff.png" % (
-                    encoded_diff_file.strip().decode('utf8'), tempfile.gettempdir(), image
-                )
-
-        self.assertTrue(test, message)
+        super()._img_diff_error(response, headers, image, max_diff=max_diff,
+                                max_size_diff=max_size_diff,
+                                unittest_data_path='qgis_server_accesscontrol')
```
Comment (pblottiere) [inline tests/src/python/test_qgsserver_accesscontrol.py]: :+1:

File: src/server/services/wms/qgswmsrenderer.cpp (hunk 1 above)
Comment (elpaso) [inline src/server/services/wms/qgswmsrenderer.cpp]:
```suggestion
        // Reverse order of layers from a group
```

File: tests/src/python/test_qgsserver_wms_getmap.py [hunk 3]
```diff
@@ -41,9 +41,10 @@
 
 
 class TestQgsServerWMSGetMap(QgsServerTestBase):
-
     """QGIS Server WMS Tests for GetMap request"""
 
+    #regenerate_reference = True
```
Comment (pblottiere) [inline tests/src/python/test_qgsserver_wms_getmap.py]: What is the purpose of `regenerate_reference`?

**Comment by pblottiere** [comment]
It looks good @elpaso, thanks!

File: tests/src/python/test_qgsserver_wms_getmap.py (hunk 3 above)
Comment (elpaso) [reply tests/src/python/test_qgsserver_wms_getmap.py]: @pblottiere https://github.com/qgis/QGIS/blob/master/tests/src/python/test_qgsserver.py#L65

Comment (pblottiere) [reply tests/src/python/test_qgsserver_wms_getmap.py]: Ah yes, I forgot that. Thanks!

**Pull request merged**